misc/peer-store/src/behaviour.rs

```diff
@@ -0,0 +1,233 @@
+use std::{
+    collections::{HashSet, VecDeque},
+    task::Poll,
+    time::Duration,
+};
+
+use futures_timer::Delay;
+use futures_util::FutureExt;
+use libp2p_core::{Multiaddr, PeerId};
+use libp2p_swarm::{dummy, FromSwarm, NetworkBehaviour};
+
+use crate::store::{AddressSource, Store};
+
+/// Events generated by [`Behaviour`] and emitted back to the [`libp2p_swarm::Swarm`].
+pub enum Event {
+    /// The peer's record has been updated.  
+    /// Manually updating a record will always emit this event
+    /// even if it provides no new information.
+    RecordUpdated { peer: PeerId },

```

Comment:
I am not sure how valuable this event is to a user, if it doesn't include any details about what was added. Right now we only emit this even when a new address was added, right? Should we then include the address here?

Maybe it also make sense to have an `Event` type parameter on the `Store` trait, and then have a variant here `Event<E> { Store { peer: PeerId, event: E }, .. }`. Wdyt?

Reply:
The thing is, where do you handle the events? The behaviour is generic over `Store` so we can't really do much on the behaviour side other than propergating them to the application level through `ToSwarm`. Having this flexibility can be of help in the future though, especially with `poll()`.